apps/wallet/src/ui/app/staking/stake/StakeForm.tsx

```diff
@@ -46,48 +49,111 @@ function StakeForm({
 
     const [formatted, symbol] = useFormatCoin(coinBalance, coinType);
 
+    const allCoins = useAppSelector(accountCoinsSelector);
+    const allCoinsOfTransferType = useMemo(
+        () => allCoins.filter((aCoin) => aCoin.type === coinType),
+        [allCoins, coinType]
+    );
+    const [coinDecimals] = useCoinDecimals(coinType);
+    const gasEstimationUnformatted = useMemo(() => {
+        return Coin.computeGasBudgetForPay(
+            allCoinsOfTransferType,
+            parseAmount(amount, coinDecimals)
+        );
+    }, [allCoinsOfTransferType, amount, coinDecimals]);
+
+    const [gasBudgetEstimation] = useFormatCoin(
+        gasEstimationUnformatted,
+        coinType
+    );
+
+    const maxToken = formatBalance(coinBalance, coinDecimals);
```

Comment:
Why you don't use useFormatCoin?
Should we subtract the `gasBudgetEstimation` from `coinBalance`?
(This is async also so we should handle it)